sample-check/src/test/samples/cli-with-working-directory-and-change-directory/multi-step/multi-step.sample.conf

```diff
@@ -0,0 +1,13 @@
+commands: [{
+    executable = "mkdir"
+    execution-subdirectory = "workDir"
+    args = "demo"
+}, {
+   executable = "cd"
+   args = "demo"
+}, {
+    executable = "bash"
+    execution-subdirectory = "workDir/demo"
```

Comment:
If I read this config file, I would expect `execution-subdirectory` to be resolved relative to the current directory, that is, I would expect the `bash` command to be run from `demo/workDir/demo`.

Another strategy, which might be better, would be to remove the `execution-subdirectory` attribute entirely and require an explicit `cd` command. This way, the sample better documents the steps a human would need to take, rather than encoding an implicit `cd` command in some steps.

Reply:
Between 0.7.0 and 0.8.0 there was this behavior change. This effectively cause all sample test in `gradle/gradle` to be broken. My thinking was to fix this breaking change and reevaluate how this is supposed to work. I understand it make sense, but it also break everything else. There should have being some sort of migration path or notice in the read me or an effort to migrate all sample test in `gradle/gradle`. We are stuck in the sample work because we can't use the latest version of the exemplar plugin to extra the steps (or expend on the step extraction code).

I see your point, I will just go fix `gradle/gradle` like it should have being when the breaking change was introduced.

Reply:
I have to say I don't really agree with the comment that `execution-subdirectory` should be seen as `cd` command. I feel it should be renamed into `working-directory` relative to the sample working directory. The behavior change with [0.8.0 is completely incompatible with all the assumption written in `gradle/gradle`](https://github.com/gradle/gradle/pull/11115/files) and most likely all the assumption everyone else has being working from. If you need a `cd`-like behavior, then we should use such command. Since I started to work on fixing `gradle/gradle` to work with this new behavior it needed a change like the following: https://github.com/gradle/gradle/pull/11115/files?file-filters%5B%5D=.java#diff-b892219573f4828ed3c88d5893d83d06L40. I feel this whole thing need to be rethought.

Reply:
I talked with Sterling and I think there were a misunderstanding of the previous comment. We are going to change the exemplar plugin to have a `baseDirectory` where `cd` command will influence and the `execution-subdirectory`, if present, will be appended `baseDirectory` to create the `workingDirectory` instead of both of them influencing the `workingDirectory`. This would make fix the behavior to be like it was on before for `gradle/gradle`.

Reply:
Just to follow-up here... Daniel and I talked about this and there was a misunderstanding.  The `cd` command should change the current working directory.  The the "execution-subdirectory" just adds to the working directory for the given command.

Reply:
I did the change.